### PR TITLE
lact (Linux AMDGPU Controller): new, 0.5.4

### PR DESCRIPTION
**Pull request by Sharelter**
Topic Description
-----------------
blueprint-compiler: new, 0.12.0
lact: new, 0.54.0

Package(s) Affected
-------------------
blueprint-compiler
lact

Security Update?
----------------
NO

Build Order
-----------

```
blueprint-compiler lact
```

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`
- [ ] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

Architectural progress for secondary ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

**Experimental Architectures**

Architectural progress for experimental ports does not impede on merging of this topic.

- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

**Comment by Sharelter** [comment]
image
image
image
image

File: app-admin/lact/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,11 @@
+PKGNAME=lact
+PKGSEC=admin
+PKGDEP="libdrm hwdata gtk-4"
+BUILDDEP="cargo libdrm hwdata gtk-4 llvm blueprint-compiler"
+USECLANG=1
+# FIXME: These architectures are not currently available ld.lld
```
Comment (MingcongBai) [inline app-admin/lact/autobuild/defines]: FIXME: ld.lld is not yet available for these architectures.

File: app-admin/lact/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,11 @@
+PKGNAME=lact
+PKGSEC=admin
+PKGDEP="libdrm hwdata gtk-4"
+BUILDDEP="cargo libdrm hwdata gtk-4 llvm blueprint-compiler"
```
Comment (MingcongBai) [inline app-admin/lact/autobuild/defines]: Do not list duplicate dependencies in BUILDDEP.

Comment (MingcongBai) [reply app-admin/lact/autobuild/defines]: Use rustc as dependency, not cargo.

File: app-admin/lact/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+USECLANG=1
+# FIXME: These architectures are not currently available ld.lld
+NOLTO__LOONGSON3=1
+NOLTO__MIPS64R6EL=1
+NOLTO__LOONGARCH64=1
+ABTYPE=plainmake
```
Comment (MingcongBai) [inline app-admin/lact/autobuild/defines]: We don't have this any more. This is probably a "rust" build type source.

File: app-admin/lact/spec [hunk 4]
```diff
@@ -0,0 +1,3 @@
+VER=0.5.4
+SRCS="git::commit=tags/v$VER::https://github.com/ilya-zlobintsev/LACT"
+CHKSUMS="SKIP"
```
Comment (MingcongBai) [inline app-admin/lact/spec:3]: Missing CHKUPDATE.

File: app-devel/blueprint-compiler/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=blueprint-compiler
+PKGSEC=devel
+ABSPLITDBG=0
```
Comment (MingcongBai) [inline app-devel/blueprint-compiler/autobuild/defines]: Dependencies? Also why disable this?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-devel/blueprint-compiler/autobuild/defines (hunk 5 above)
Comment (Sharelter) [reply app-devel/blueprint-compiler/autobuild/defines]: About `ABSPLITDBG=0`
I got this error:
```
[WARN]:  QA (W322): Zero-byte files found:

/var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler/abdist/usr/lib/python3.10/site-packages/blueprintcompiler/__init__.py

[WARN]:  QA warning(s) found, please refer to abqawarn.log ...
[INFO]:  Packing dpkg package(s) ...
[DEBUG]: Member 'mainline' in Architecture group 'mainline' mismatches retro.
[INFO]:  All architecture group members 'mainline' mismatches retro: taking false branch.
[INFO]:  Architecture loongarch64 mismatches mips32r6el: taking false branch.
[INFO]:  Architecture loongarch64 mismatches riscv64: taking false branch.
dpkg-deb: building package 'blueprint-compiler' in 'blueprint-compiler_0.12.0-0_loongarch64.deb'.
In file included from /usr/bin/autobuild:13
/usr/lib/autobuild4/proc/80-pm-pack-install.sh:19: In function `source':
/usr/lib/autobuild4/proc/80-pm-pack-install.sh:19: error: command exited with 0
   18 |         # A empty dir or dir with no regular files becomes a fatal error if ABSPLITDBG is set.
>  19 |         abdie "ABSPLITDBG is set, but we can't find any symbol files in ${SYMDIR}"
   20 |     fi

autobuild encountered an error and couldn't continue.
ABSPLITDBG is set, but we can't find any symbol files in /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler/abdist-dbg
------------------------------autobuild 4.1.9------------------------------
Go to ‘https://github.com/AOSC-Dev/autobuild4’ for more information on this error.
[ERROR]: Build directory of the failed package: /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/acbs/main.py", line 280, in build_sequential
    invoke_autobuild(task, build_dir, scoped_stage2)
  File "/usr/lib/python3.10/site-packages/acbs/utils.py", line 222, in invoke_autobuild
    start_build_capture(env_dict, build_dir)
  File "/usr/lib/python3.10/site-packages/acbs/utils.py", line 168, in start_build_capture
    raise RuntimeError('autobuild3 did not exit successfully.')
RuntimeError: autobuild3 did not exit successfully.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/acbs/main.py", line 149, in build
    self.build_sequential(build_timings, packages)
  File "/usr/lib/python3.10/site-packages/acbs/main.py", line 286, in build_sequential
    raise RuntimeError(
RuntimeError: Build directory of the failed package: /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler
```

Comment (MingcongBai) [reply app-devel/blueprint-compiler/autobuild/defines]: > I got this error:
> 
> ```
> [WARN]:  QA (W322): Zero-byte files found:
> 
> /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler/abdist/usr/lib/python3.10/site-packages/blueprintcompiler/__init__.py
> 
> [WARN]:  QA warning(s) found, please refer to abqawarn.log ...
> [INFO]:  Packing dpkg package(s) ...
> [DEBUG]: Member 'mainline' in Architecture group 'mainline' mismatches retro.
> [INFO]:  All architecture group members 'mainline' mismatches retro: taking false branch.
> [INFO]:  Architecture loongarch64 mismatches mips32r6el: taking false branch.
> [INFO]:  Architecture loongarch64 mismatches riscv64: taking false branch.
> dpkg-deb: building package 'blueprint-compiler' in 'blueprint-compiler_0.12.0-0_loongarch64.deb'.
> In file included from /usr/bin/autobuild:13
> /usr/lib/autobuild4/proc/80-pm-pack-install.sh:19: In function `source':
> /usr/lib/autobuild4/proc/80-pm-pack-install.sh:19: error: command exited with 0
>    18 |         # A empty dir or dir with no regular files becomes a fatal error if ABSPLITDBG is set.
> >  19 |         abdie "ABSPLITDBG is set, but we can't find any symbol files in ${SYMDIR}"
>    20 |     fi
> 
> autobuild encountered an error and couldn't continue.
> ABSPLITDBG is set, but we can't find any symbol files in /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler/abdist-dbg
> ------------------------------autobuild 4.1.9------------------------------
> Go to ‘https://github.com/AOSC-Dev/autobuild4’ for more information on this error.
> [ERROR]: Build directory of the failed package: /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler
> Traceback (most recent call last):
>   File "/usr/lib/python3.10/site-packages/acbs/main.py", line 280, in build_sequential
>     invoke_autobuild(task, build_dir, scoped_stage2)
>   File "/usr/lib/python3.10/site-packages/acbs/utils.py", line 222, in invoke_autobuild
>     start_build_capture(env_dict, build_dir)
>   File "/usr/lib/python3.10/site-packages/acbs/utils.py", line 168, in start_build_capture
>     raise RuntimeError('autobuild3 did not exit successfully.')
> RuntimeError: autobuild3 did not exit successfully.
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/lib/python3.10/site-packages/acbs/main.py", line 149, in build
>     self.build_sequential(build_timings, packages)
>   File "/usr/lib/python3.10/site-packages/acbs/main.py", line 286, in build_sequential
>     raise RuntimeError(
> RuntimeError: Build directory of the failed package: /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler
> ```

Is this a "noarch" package?

Comment (OriginCode) [inline app-devel/blueprint-compiler/autobuild/defines:3]: There's a meson build option that compiles documentation (default is false), enable it to include docs in the package.

**Review by OriginCode — CHANGES_REQUESTED** [review]
~~There's a meson build option that compiles documentation (default is false), enable it to include docs in the package.~~

See below.

File: app-devel/blueprint-compiler/autobuild/defines (hunk 5 above)
Comment (OriginCode) [inline app-devel/blueprint-compiler/autobuild/defines]: There's a meson build option that compiles documentation (default is false), enable it to include docs in the package.

**Review by OriginCode — CHANGES_REQUESTED** [review]

File: app-devel/blueprint-compiler/autobuild/defines (hunk 5 above)
Comment (Sharelter) [reply app-devel/blueprint-compiler/autobuild/defines]: Did not know what PKGDEP it needs
and BUILDDEP is included in the BuiltKit

Comment (Sharelter) [reply app-devel/blueprint-compiler/autobuild/defines]: > > I got this error:
> > ```
> > [WARN]:  QA (W322): Zero-byte files found:
> > 
> > /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler/abdist/usr/lib/python3.10/site-packages/blueprintcompiler/__init__.py
> > 
> > [WARN]:  QA warning(s) found, please refer to abqawarn.log ...
> > [INFO]:  Packing dpkg package(s) ...
> > [DEBUG]: Member 'mainline' in Architecture group 'mainline' mismatches retro.
> > [INFO]:  All architecture group members 'mainline' mismatches retro: taking false branch.
> > [INFO]:  Architecture loongarch64 mismatches mips32r6el: taking false branch.
> > [INFO]:  Architecture loongarch64 mismatches riscv64: taking false branch.
> > dpkg-deb: building package 'blueprint-compiler' in 'blueprint-compiler_0.12.0-0_loongarch64.deb'.
> > In file included from /usr/bin/autobuild:13
> > /usr/lib/autobuild4/proc/80-pm-pack-install.sh:19: In function `source':
> > /usr/lib/autobuild4/proc/80-pm-pack-install.sh:19: error: command exited with 0
> >    18 |         # A empty dir or dir with no regular files becomes a fatal error if ABSPLITDBG is set.
> > >  19 |         abdie "ABSPLITDBG is set, but we can't find any symbol files in ${SYMDIR}"
> >    20 |     fi
> > 
> > autobuild encountered an error and couldn't continue.
> > ABSPLITDBG is set, but we can't find any symbol files in /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler/abdist-dbg
> > ------------------------------autobuild 4.1.9------------------------------
> > Go to ‘https://github.com/AOSC-Dev/autobuild4’ for more information on this error.
> > [ERROR]: Build directory of the failed package: /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler
> > Traceback (most recent call last):
> >   File "/usr/lib/python3.10/site-packages/acbs/main.py", line 280, in build_sequential
> >     invoke_autobuild(task, build_dir, scoped_stage2)
> >   File "/usr/lib/python3.10/site-packages/acbs/utils.py", line 222, in invoke_autobuild
> >     start_build_capture(env_dict, build_dir)
> >   File "/usr/lib/python3.10/site-packages/acbs/utils.py", line 168, in start_build_capture
> >     raise RuntimeError('autobuild3 did not exit successfully.')
> > RuntimeError: autobuild3 did not exit successfully.
> > 
> > During handling of the above exception, another exception occurred:
> > 
> > Traceback (most recent call last):
> >   File "/usr/lib/python3.10/site-packages/acbs/main.py", line 149, in build
> >     self.build_sequential(build_timings, packages)
> >   File "/usr/lib/python3.10/site-packages/acbs/main.py", line 286, in build_sequential
> >     raise RuntimeError(
> > RuntimeError: Build directory of the failed package: /var/cache/acbs/build/acbs.ai_gbz5c/blueprint-compiler
> > ```
> 
> Is this a "noarch" package?

Yeah, it is.

Comment (Sharelter) [reply app-devel/blueprint-compiler/autobuild/defines]: Looks it is the problem which means that it the autobuild cannot find the debug symbols

File: app-devel/blueprint-compiler/autobuild/defines [hunk 6]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=blueprint-compiler
+PKGSEC=devel
+ABHOST=noarch
+PKGDES="Blueprint language compiler"
```
Comment (MingcongBai) [inline app-devel/blueprint-compiler/autobuild/defines:3]: Dependencies???

**Review by MingcongBai — DISMISSED** [review]

File: app-admin/lact/spec (hunk 4 above)
Comment (Sharelter) [reply app-admin/lact/spec:3]: It is not traced by anyita, now trying to use github api

Comment (Sharelter) [reply app-admin/lact/spec:3]: Updated

File: app-devel/blueprint-compiler/autobuild/defines (hunk 6 above)
Comment (Sharelter) [reply app-devel/blueprint-compiler/autobuild/defines:3]: Added.

File: app-admin/lact/autobuild/defines [hunk 7]
```diff
@@ -3,6 +3,7 @@ PKGSEC=admin
 PKGDEP="libdrm hwdata gtk-4"
 BUILDDEP="rustc gtk-4 llvm blueprint-compiler"
 USECLANG=1
+MESON_AFTER=-Ddocumentation=true
```
Comment (OriginCode) [inline app-admin/lact/autobuild/defines:6]: Add double brackets for the right hand side.

File: app-admin/lact/autobuild/defines [hunk 8]
```diff
@@ -0,0 +1,12 @@
+PKGNAME=lact
+PKGSEC=admin
+PKGDEP="libdrm hwdata gtk-4"
+BUILDDEP="rustc gtk-4 llvm blueprint-compiler"
+USECLANG=1
+MESON_AFTER=-Ddocumentation=true
```
Comment (OriginCode) [inline app-admin/lact/autobuild/defines]: Add brackets to the right hand side.

**Review by OriginCode — DISMISSED** [review]

File: app-admin/lact/autobuild/defines (hunk 8 above)
Comment (Sharelter) [reply app-admin/lact/autobuild/defines]: Updated.

**Review by jiegec — APPROVED** [review]

**Pull request merged**